source/README.adoc

```diff
@@ -0,0 +1,142 @@
+   | `NS`, `NAME` and `TOPIC` are derived from the Source configuration.
+| `id` | a unique id | This uses the MessageId if available, and a UUID otherwise.
+| `subject` | the message's id | empty string if no message ID is present.
+| `datacontenttype` | `application/json` | Currently static.
+| `key` | the message's id | empty string if no message ID is present.
+|===
+
+The event's payload will be set to the data content of the message.
+
+=== Samples
+
+(in CloudEvents JSON format):
+[source,json]
+----
+{
+  "specversion": "1.0",
+  "type": "dev.knative.rabbitmq.event",
+  "source": "/apis/v1/namespaces/default/rabbitmqsources/rabbitmq-source",
+  "id": "f00c1f52-33a1-4d3d-993f-750f20c804da",
+  "time": "2020-12-18T01:15:20.450860898Z",
+  "subject": "",
+  "datacontenttype": "application/json",
+  "key": "",
+  "data": "Hello rabbitmq!"
+}
+----
+
+== Creating and Managing Sources
+
+
+.Table Source Parameters
+|===
+| Field | Value
+
+| `spec.brokers` | Host+Port of the Broker, with a trailing "/"
+| `user.secretKeyRef` | Username for broker auth; field `key` in a Kubernetes Secret named `name`
+| `password.secretKeyRef` | Password for broker auth; field `key` in a Kubernetes Secret named `name`
+| `topic` | The topic for the exchange
+| `exchange_config` | Settings for the exchange
+| `exchange_config.type` | https://www.rabbitmq.com/tutorials/amqp-concepts.html#exchanges[Exchange type]. Can be "fanout", "direct", "topic", "match" or "headers"
+| `exchange_config.durable` | Boolean
+| `exchange_config.auto_deleted` | Boolean
+| `exchange_config.internal` | Boolean
+| `exchange_config.nowait` | Boolean
+| `queue_config` | Settings for the queue
+| `queue_config.name` | Name of the queue (may be empty)
+| `queue_config.routing_key` | Routing key for the queue
+| `queue_config.durable` | Boolean
+| `queue_config.delete_when_unused` | Boolean
+| `queue_config.exclusive` | Boolean
+| `queue_config.nowait` | Boolean
+| `sink` | A reference to an https://knative.dev/docs/eventing/#event-consumers[Addressable] Kubernetes object.
+|===
+
+The following example object assumes a secret created with the following:
+----
+kubectl create secret generic rabbitmq-secret \
+  --from-literal=user=rabbit-user \
+  --from-file=password=/tmp/password
+----
+
+Note that not all parameters need to be specified; unspecified parameters will be
+defaulted false / empty.
+
+[source,yaml]
+----
+apiVersion: sources.knative.dev/v1alpha1
+kind: RabbitmqSource
+metadata:
+  name: rabbitmq-source
+spec:
+  brokers: "rabbitmq:5672/"
+  user:
+    secretKeyRef:
+      name: "rabbitmq-secret"
+      key: "user"
+  password:
+    secretKeyRef:
+      name: "rabbitmq-secret"
+      key: "password"
+  exchange_config:
+    type: "fanout"
+    durable: true
+    auto_deleted: false
+  sink:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: event-display
+----
+
+== Administration
```

Comment:
Not sure I like this in the one file, I think the idea was more that there would be an introduction "guide" with conceptual info etc and then an Admin guide with all of this info as a separate page. Also, I would avoid having any heading that is directly followed by another heading with nothing between; there should be some paragraph or something here explaining what this admin section is